### PR TITLE
optimizing flow tests by seperating and parallelizing test suites

### PR DESCRIPTION
**Pull request by heavycrystal**
1. Split each connector into its own test suite, and rewrote them to allow running multiple test suites in parallel.
2. Better separation on source PG instance [shared by all test suites of a run] and destination BQ and SF instances [shared by all runs].
3. some code cleanup and fixing some tests up.
4. Switched to using a vanilla PG instance, but configured with higher logical replication limits.

### CODE REVIEW

File: flow/e2e/test_utils.go [hunk 1]
```diff
@@ -24,3 +38,277 @@
...
+	}
+
+	// generate a 20 MB json and update id[0]'s col f5 to it
+	v, err := generate20MBJson()
+	if err != nil {
+		return err
+	}
+	_, err = pool.Exec(context.Background(), fmt.Sprintf(`
+		UPDATE e2e_test_%s.%s SET f5 = '%s' WHERE id = '%s';
+	`, suffix, tableName, v, ids[0]))
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/e2e/test_utils.go:216]: fixed

File: flow/e2e/postgres/qrep_flow_pg_test.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+// Implement TearDownAllSuite interface to tear down the test suite
+func (s *PeerFlowE2ETestSuitePG) TearDownSuite() {
+	err := e2e.TearDownPostgres(s.pool, postgresSuffix)
+	if err != nil {
+		s.Fail("failed to drop Postgres schema", err)
+	}
+}
+
+func (s *PeerFlowE2ETestSuitePG) setupSourceTable(tableName string, rowCount int) {
+	e2e.CreateSourceTableQRep(s.pool, postgresSuffix, tableName)
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/e2e/postgres/qrep_flow_pg_test.go:57]: fixed

File: flow/e2e/postgres/qrep_flow_pg_test.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	env := s.NewTestWorkflowEnvironment()
+	e2e.RegisterWorkflowsAndActivities(env)
+
+	numRows := 10
+
+	srcTable := "test_qrep_flow_avro_pg_1"
+	s.setupSourceTable(srcTable, numRows)
+
+	dstTable := "test_qrep_flow_avro_pg_2"
+	e2e.CreateSourceTableQRep(s.pool, postgresSuffix, dstTable) // the name is misleading, but this is the destination table
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/e2e/postgres/qrep_flow_pg_test.go:109]: fixed

File: flow/e2e/snowflake/qrep_flow_sf_test.go [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+	connpostgres "github.com/PeerDB-io/peer-flow/connectors/postgres"
+	"github.com/PeerDB-io/peer-flow/e2e"
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+)
+
+func (s *PeerFlowE2ETestSuiteSF) setupSourceTable(tableName string, rowCount int) {
+	e2e.CreateSourceTableQRep(s.pool, snowflakeSuffix, tableName)
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/e2e/snowflake/qrep_flow_sf_test.go:15]: fixed

File: flow/e2e/bigquery/qrep_flow_bq_test.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+	connpostgres "github.com/PeerDB-io/peer-flow/connectors/postgres"
+	"github.com/PeerDB-io/peer-flow/e2e"
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/stretchr/testify/require"
+)
+
+func (s *PeerFlowE2ETestSuiteBQ) setupSourceTable(tableName string, rowCount int) {
+	e2e.CreateSourceTableQRep(s.pool, bigquerySuffix, tableName)
+	e2e.PopulateSourceTable(s.pool, bigquerySuffix, tableName, rowCount)
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/e2e/bigquery/qrep_flow_bq_test.go:15]: fixed

File: flow/e2e/s3/qrep_flow_s3_test.go [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+	}
+
+	env := s.NewTestWorkflowEnvironment()
+	e2e.RegisterWorkflowsAndActivities(env)
+
+	jobName := "test_complete_flow_s3"
+	schemaQualifiedName := fmt.Sprintf("e2e_test_%s.%s", s3Suffix, jobName)
+
+	s.setupSourceTable(jobName, 10)
+	query := fmt.Sprintf("SELECT * FROM %s WHERE updated_at >= {{.start}} AND updated_at < {{.end}}", schemaQualifiedName)
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/e2e/s3/qrep_flow_s3_test.go:97]: fixed

**Review by iskakaushik — APPROVED** [review]

File: flow/e2e/snowflake/qrep_flow_sf_test.go [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+	e2e.PopulateSourceTable(s.pool, snowflakeSuffix, tableName, rowCount)
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/e2e/snowflake/qrep_flow_sf_test.go]: fixed

File: flow/e2e/postgres/qrep_flow_pg_test.go [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+	return rows.Err()
+}
+
+func (s *PeerFlowE2ETestSuitePG) Test_Complete_QRep_Flow_Multi_Insert_PG() {
+	env := s.NewTestWorkflowEnvironment()
+	e2e.RegisterWorkflowsAndActivities(env)
+
+	numRows := 10
+
+	srcTable := "test_qrep_flow_avro_pg_1"
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/e2e/postgres/qrep_flow_pg_test.go:107]: false positive

File: flow/e2e/s3/qrep_flow_s3_test.go [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+	pool     *pgxpool.Pool
+	s3Helper *S3TestHelper
+}
+
+func TestPeerFlowE2ETestSuiteS3(t *testing.T) {
+	suite.Run(t, new(PeerFlowE2ETestSuiteS3))
+}
+
+func (s *PeerFlowE2ETestSuiteS3) setupSourceTable(tableName string, rowCount int) {
+	e2e.CreateSourceTableQRep(s.pool, s3Suffix, tableName)
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/e2e/s3/qrep_flow_s3_test.go]: fixed

File: flow/e2e/s3/qrep_flow_s3_test.go [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+	e2e.PopulateSourceTable(s.pool, s3Suffix, tableName, rowCount)
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/e2e/s3/qrep_flow_s3_test.go]: fixed

**Pull request merged**